### PR TITLE
[CWS] Rework secprofile metrics

### PR DESCRIPTION
**Pull request by spikat**
### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/security/security_profile/profile/manager.go [hunk 1]
```diff
@@ -34,42 +34,67 @@ import (
...
-// DefaultProfileName used as default profile name
-const DefaultProfileName = "default"
-
-func (efr EventFilteringResult) toTag() string {
+func (efr EventFilteringProfileState) toTag() string {
 	switch efr {
 	case NoProfile:
 		return fmt.Sprintf("in_profile:no_profile")
+	case UnstableProfile:
+		return fmt.Sprintf("in_profile:unstable_profile")
```
Comment (Gui774ume) [inline pkg/security/security_profile/profile/manager.go:62]: "profile_state" ?

**Review by Gui774ume — APPROVED** [review]
Tiny comment but otherwise lgtm

File: pkg/security/security_profile/profile/manager.go (hunk 1 above)
Comment (spikat) [reply pkg/security/security_profile/profile/manager.go]: Good catch, thanks! Fixed.

**Review by lebauce — APPROVED** [review]

**Review by safchain — APPROVED** [review]

File: pkg/security/security_profile/profile/manager.go [hunk 2]
```diff
@@ -34,44 +33,69 @@ import (
...
+	// WorkloadWarmup is used to count the learned events due to workload warm up time
 	WorkloadWarmup
 )
 
-// DefaultProfileName used as default profile name
-const DefaultProfileName = "default"
+func (efr EventFilteringProfileState) toTag() string {
+	switch efr {
+	case NoProfile:
+		return fmt.Sprintf("profile_state:no_profile")
```
Comment (safchain) [inline pkg/security/security_profile/profile/manager.go:59]: nit: Sprintf is useless no ?

File: pkg/security/security_profile/profile/manager.go [hunk 3]
```diff
@@ -34,44 +33,69 @@ import (
...
+// EventFilteringResult is used to compute metrics for the event filtering feature
+type EventFilteringResult uint8
+
+const (
+	// InProfile is used to count the events that matched a profile
+	InProfile EventFilteringResult = iota
+	// NotInProfile is used to count the events that didn't match their profile
+	NotInProfile
+	// Not applicable, for profil NoProfile and UnstableProfile state
+	NA
```
Comment (safchain) [inline pkg/security/security_profile/profile/manager.go:81]: Should we put this as the first value ? so that `NA` is equal to 0

**Review by safchain — APPROVED** [review]
few nits, otherwise LGTM

**Comment by spikat** [comment]
> few nits, otherwise LGTM

thanks! suggestions applied

**Pull request merged**